### PR TITLE
Map NetworkState.

### PR DESCRIPTION
**Pull request by Fuyukai**
This seems like a rather round-about way to define a map.

### CODE REVIEW

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 1]
```diff
@@ -1,6 +1,22 @@
 CLASS net/minecraft/class_2539 net/minecraft/network/NetworkState
+	CLASS class_4532 PacketHandlerMap
+		FIELD field_20596 packetIds Lit/unimi/dsi/fastutil/objects/Object2IntMap;
+		FIELD field_20597 packetSuppliers Ljava/util/List;
+		METHOD method_22309 iterate ()Ljava/lang/Iterable;
+		METHOD method_22310 getPacketFromID (I)Lnet/minecraft/class_2596;
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
		METHOD method_22310 createPacket (I)Lnet/minecraft/class_2596;
```

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 2]
```diff
@@ -1,6 +1,22 @@
 CLASS net/minecraft/class_2539 net/minecraft/network/NetworkState
+	CLASS class_4532 PacketHandlerMap
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
	CLASS class_4532 PacketHandler
```
or `PacketReceiver`

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+		FIELD field_20596 packetIds Lit/unimi/dsi/fastutil/objects/Object2IntMap;
+		FIELD field_20597 packetSuppliers Ljava/util/List;
+		METHOD method_22309 iterate ()Ljava/lang/Iterable;
+		METHOD method_22310 getPacketFromID (I)Lnet/minecraft/class_2596;
+			ARG 1 id
+		METHOD method_22312 getIdFromPacket (Ljava/lang/Class;)Ljava/lang/Integer;
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
		METHOD method_22312 getId (Ljava/lang/Class;)Ljava/lang/Integer;
```

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 4]
```diff
@@ -1,6 +1,22 @@
 CLASS net/minecraft/class_2539 net/minecraft/network/NetworkState
+	CLASS class_4532 PacketHandlerMap
+		FIELD field_20596 packetIds Lit/unimi/dsi/fastutil/objects/Object2IntMap;
+		FIELD field_20597 packetSuppliers Ljava/util/List;
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
		FIELD field_20597 packetFactories Ljava/util/List;
```

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 5]
```diff
@@ -1,6 +1,22 @@
...
+		METHOD method_22309 iterate ()Ljava/lang/Iterable;
+		METHOD method_22310 getPacketFromID (I)Lnet/minecraft/class_2596;
+			ARG 1 id
+		METHOD method_22312 getIdFromPacket (Ljava/lang/Class;)Ljava/lang/Integer;
+			ARG 1 packet
+		METHOD method_22313 registerPacket (Ljava/lang/Class;Ljava/util/function/Supplier;)Lnet/minecraft/class_2539$class_4532;
+			ARG 1 packetClass
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
			ARG 1 type
```

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 6]
```diff
@@ -1,6 +1,22 @@
...
+			ARG 2 packetSupplier
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
			ARG 2 factory
```

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 7]
```diff
@@ -1,6 +1,22 @@
...
+	CLASS class_4533 PacketHandlerMapBuilder
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
	CLASS class_4533 PacketHandlerInitializer
```

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 8]
```diff
@@ -1,6 +1,22 @@
...
+		METHOD method_22315 putPacketHandlerMap (Lnet/minecraft/class_2598;Lnet/minecraft/class_2539$class_4532;)Lnet/minecraft/class_2539$class_4533;
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
		METHOD method_22315 setup (Lnet/minecraft/class_2598;Lnet/minecraft/class_2539$class_4532;)Lnet/minecraft/class_2539$class_4533;
```

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 9]
```diff
@@ -10,3 +26,4 @@ CLASS net/minecraft/class_2539 net/minecraft/network/NetworkState
 	METHOD method_10785 getId ()I
 	METHOD method_10786 getPacketHandlerState (Lnet/minecraft/class_2596;)Lnet/minecraft/class_2539;
 		ARG 0 handler
+	METHOD method_22308 getPacketHandlerMapBuilder ()Lnet/minecraft/class_2539$class_4533;
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
	METHOD method_22308 getPacketHandlerInitializer ()Lnet/minecraft/class_2539$class_4533;
```

**Review by liach — COMMENTED** [review]
IMO You have misunderstood some code you've mapped.

**Comment by Fuyukai** [comment]
Okay, that makes more sense than what I initially thought it was. Although, with this, it seems like it would make more sense to rename getPacketHandler to createPacket (I didn't want to mess with existing mappings) too.

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 10]
```diff
@@ -1,6 +1,23 @@
 CLASS net/minecraft/class_2539 net/minecraft/network/NetworkState
+	CLASS class_4532 PacketHandler
+		FIELD field_20596 packetIds Lit/unimi/dsi/fastutil/objects/Object2IntMap;
+		FIELD field_20597 packetFactories Ljava/util/List;
+		METHOD method_22309 iterate ()Ljava/lang/Iterable;
+		METHOD method_22310 createPacket (I)Lnet/minecraft/class_2596;
+			ARG 1 id
+		METHOD method_22312 getId (Ljava/lang/Class;)Ljava/lang/Integer;
+			ARG 1 packet
+		METHOD method_22313 registerPacket (Ljava/lang/Class;Ljava/util/function/Supplier;)Lnet/minecraft/class_2539$class_4532;
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
		METHOD method_22313 register (Ljava/lang/Class;Ljava/util/function/Supplier;)Lnet/minecraft/class_2539$class_4532;
```

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 11]
```diff
@@ -1,6 +1,23 @@
 CLASS net/minecraft/class_2539 net/minecraft/network/NetworkState
+	CLASS class_4532 PacketHandler
+		FIELD field_20596 packetIds Lit/unimi/dsi/fastutil/objects/Object2IntMap;
+		FIELD field_20597 packetFactories Ljava/util/List;
+		METHOD method_22309 iterate ()Ljava/lang/Iterable;
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
		METHOD method_22309 getPacketTypes ()Ljava/lang/Iterable;
```

File: mappings/net/minecraft/network/NetworkState.mapping [hunk 12]
```diff
@@ -10,3 +27,4 @@ CLASS net/minecraft/class_2539 net/minecraft/network/NetworkState
 	METHOD method_10785 getId ()I
 	METHOD method_10786 getPacketHandlerState (Lnet/minecraft/class_2596;)Lnet/minecraft/class_2539;
 		ARG 0 handler
+	METHOD method_22308 getPacketHandlerIniitializer ()Lnet/minecraft/class_2539$class_4533;
```
Comment (liach) [inline mappings/net/minecraft/network/NetworkState.mapping]:
```suggestion
	METHOD method_22308 createPacketHandlerInitializer ()Lnet/minecraft/class_2539$class_4533;
```

**Review by liach — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**